WS2012R2/lisa/setupscripts/FCOPY_negative.ps1

```diff
@@ -0,0 +1,271 @@
+
+<#
+.Synopsis
+    This script tests the file copy negative functionality test.
+
+.Description
+    The script will verify fail to copy a random generated 10MB file from Windows host to
+	the Linux VM, when target folder is immutable, 'Guest Service Interface' disabled and
+	hyperverfcopyd is disabled.
+
+    A typical XML definition for this test case would look similar
+    to the following:
+	<test>
+		<testName>FCOPY_negative</testName>
+		<testScript>setupscripts\FCOPY_negative.ps1</testScript>
+		<timeout>900</timeout>
+		<testParams>
+			<param>TC_COVERED=FCopy-08</param>
+		</testParams>
+		<noReboot>False</noReboot>
+	</test>
+
+.Parameter vmName
+    Name of the VM to test.
+
+.Parameter hvServer
+    Name of the Hyper-V server hosting the VM.
+
+.Parameter testParams
+    Test data for this test case.
+
+.Example
+    setupScripts\FCOPY_negative.ps1 -vmName NameOfVm -hvServer localhost -testParams 'sshKey=path/to/ssh;ipv4=ipaddress'
+#>
+
+param([string] $vmName, [string] $hvServer, [string] $testParams)
+
+$retVal = $false
+$testfile = $null
+
+#######################################################################
+#
+#	Main body script
+#
+#######################################################################
+
+# Checking the input arguments
+if (-not $vmName) {
+    "Error: VM name is null!"
+    return $retVal
+}
+
+if (-not $hvServer) {
+    "Error: hvServer is null!"
+    return $retVal
+}
+
+if (-not $testParams) {
+    "Error: No testParams provided!"
+    "This script requires the test case ID and VM details as the test parameters."
+    return $retVal
+}
+
+#
+# Checking the mandatory testParams. New parameters must be validated here.
+#
+$params = $testParams.Split(";")
+foreach ($p in $params) {
+    $fields = $p.Split("=")
+
+    if ($fields[0].Trim() -eq "TC_COVERED") {
+        $TC_COVERED = $fields[1].Trim()
+    }
+	if ($fields[0].Trim() -eq "rootDir") {
+        $rootDir = $fields[1].Trim()
+    }
+	if ($fields[0].Trim() -eq "ipv4") {
+		$IPv4 = $fields[1].Trim()
+    }
+	if ($fields[0].Trim() -eq "sshkey") {
+        $sshkey = $fields[1].Trim()
+    }
+}
+
+#
+# Change the working directory for the log files
+# Delete any previous summary.log file, then create a new one
+#
+if (-not (Test-Path $rootDir)) {
+    "Error: The directory `"${rootDir}`" does not exist"
+    return $retVal
+}
+cd $rootDir
+
+# Source TCUtils.ps1
+if (Test-Path ".\setupScripts\TCUtils.ps1") {
+    . .\setupScripts\TCUtils.ps1
+} else {
+    "Error: Could not find setupScripts\TCUtils.ps1"
+}
+
+# If host build number lower than 9600, skip test
+$BuildNumber = GetHostBuildNumber $hvServer
+if ($BuildNumber -eq 0){
+    return $false
+}
+elseif ($BuildNumber -lt 9600){
+    return $Skipped
+}
+
+# Source FCOPY_Utils.ps1
+. .\setupScripts\FCOPY_utils.ps1
+
+# If vm does not support systemd, skip test.
+$sts = Check-Systemd
+if ($sts[-1] -eq $false){
+	return $Skipped
```

Comment:
can you please add a logged message here that "Distro does not support systemd, skipping test."?
That way we know directly the reason when reviewing the results.

Reply:
Done